### ISSUE TITLE
Estimating uncertainty on fitted parameters

### ISSUE THREAD

**Issue by tritemio** [scikit-optimize/scikit-optimize#706]
First, let me say thanks for this great package!

I am using the Bayesian optimization `gp_minimize` to find the minimum of an hard-to-compute function.

At the end of the optimization I obtain "best-fit" parameters. I'd like to estimate the uncertainty on each of them. What is the recommended approach in scikit-optimize?

The function evaluations performed during the optimization are used to compute an Gaussian-process approximation of the loss function. I was thinking that, if there was a way to extract this analytical approximation, I could compute the second derivative with respect to each parameter as an indication of "uncertainty". Is this possible? Are there other better ways?

Thanks for any suggestion.

**Comment by iaroslav-ai** [comment]
Hi, we do not have any functionality that evaluates uncertainty on specific parameters. Maybe what you suggest is the way to go. 

What do you expect the uncertainty value for a particular value of parameter to tell you? The likelihood that a given point is closer than some epsilon to the global optmum? Or would it be something else? Maybe if you could provide a use case where you would use the uncertainty values we could give some suggestions.

**Comment by tritemio** [comment]
Hi @iaroslav-ai,

I am using scikit-optimize to fit parameters of a single-molecule dynamics simulators to experimental results. The loss function is basically a metric of how much the simulation differs from the experiment.
Parameters are physical properties of the molecule. I would like to provide an estimation of the uncertainty of the fit. For example, for a given noise level with variance sigma^2, I could estimate how much to vary a parameter from the minimum to increase the loss function more than 1 sigma. Smaller variations of the parameter are not statistical significant, meaning that with the current information we cannot tell where the true value lies within the range.

Does this make sense?

**Comment by iaroslav-ai** [comment]
Hi,
sorry for a delayed response! 
Firstly, that is a cool use - case :D Yes, that makes sense to me.
Given the description of what you want, I think the way that you suggested is the right way. What you look for is the `.models` property of optimization function [result](https://github.com/scikit-optimize/scikit-optimize/blob/000b9d848c4afd5cc12852c3df0b1b76476d00b4/skopt/utils.py#L57), which contain the list of fitted gaussian processes for every iteration of the algorithm. You can take the -1st, which would be the most recent / accurate one. Then, you can try to vary different parameters, and see what the estimated objective would be. Note that the model is [an estimator, similar to sklearn estimators](https://github.com/scikit-optimize/scikit-optimize/blob/000b9d848c4afd5cc12852c3df0b1b76476d00b4/skopt/learning/gaussian_process/gpr.py#L39). Alternatively, you can also try estimating the curvature of Gaussian process. 

Beyond that I cannot suggest anything. Maybe if you find something let us know here! :)

**Comment by iaroslav-ai** [comment]
Feel free to reopen if your issue still stands.
